IPython/core/interactiveshell.py

```diff
@@ -2873,16 +2882,19 @@ def run_code(self, code_obj, result=None):
         except SystemExit as e:
             if result is not None:
                 result.error_in_exec = e
+                self.last_execution_succeeded = False
```

Comment:
`run_code` and `run_ast_nodes` return a failure flag, so I think we can set this in one place in `run_cell` rather than three places in here. As this file is now, that one place is line 2716.